clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1647,19 +1645,57 @@ void FunctionVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV) {
   auto RetVar = getOnly(returnVars);
   RetVar->mergeDeclaration(FromRetVar);
   // Transplant params.
-  assert(From->numParams() == numParams());
-  for (unsigned i = 0; i < From->numParams(); i++) {
-    CVarSet &FromP = From->getParamVar(i);
-    CVarSet &P = getParamVar(i);
-    auto FromVar = getOnly(FromP);
-    auto Var = getOnly(P);
-    Var->mergeDeclaration(FromVar);
+  handle_params(From, this,
+                [](ConstraintVariable *F, ConstraintVariable *T)
+                { T->mergeDeclaration(F); });
+}
+
+void FunctionVariableConstraint::handle_params
+(FVConstraint *From, FVConstraint *To,
+ std::function<void(ConstraintVariable*,ConstraintVariable*)> f) {
+  bool fromEmpty = From->numParams() == 0;
+  bool toEmpty = To->numParams() == 0;
```

Comment:
Just because the number of parameters is 0 doesn't mean it's a `f()` prototype; might be a `f(void)` one. What if we are attempting to merge two 'void f(void)` prototypes? I think you might need to check the non-prototype thing again?